internal/appsec/appsec.go

```diff
@@ -129,10 +130,22 @@ func (a *appsec) start() error {
 
 // Stop AppSec by unregistering the security protections.
 func (a *appsec) stop() {
-	if a.started {
-		a.started = false
-		a.unregisterWAF()
-		a.limiter.Stop()
-		a.disableRCBlocking()
+	if !a.started {
+		return
+	}
+	a.started = false
+
+	// Stop RC first so that the following is guaranteed not to be concurrent
+	// anymore.
+	a.stopRC()
```

Comment:
I think this breaks remote activation. This means that once appsec gets remote deactivated, you can't receive rc updates anymore

Reply:
74673d5536e47 puts `stopRC()` back where it was, but we now unregister the callback to stay safe wrt concurrency.
Also since we now unregister callbacks we can register them in more "accurate" places i.e at the same place where we register capabilites.